components/core/tests/test-BoundedReader.cpp

```diff
@@ -0,0 +1,99 @@
+#include <string>
+
+#include <Catch2/single_include/catch2/catch.hpp>
+
+#include "../src/clp/BoundedReader.hpp"
+#include "../src/clp/ErrorCode.hpp"
+#include "../src/clp/StringReader.hpp"
+
+TEST_CASE("Test Bounded Reader", "[BoundedReader]") {
+    constexpr char cTestString[]{"0123456789"};
```

Comment:
I'm using constexpr char[] because StringReader takes std::string const& and I don't want to manually initialize an std::string every time I create a StringReader.

Reply:
- This was to follow our guideline here: https://www.notion.so/yscope/WIP-Coding-Guidelines-9a308b847a5343958ba3cb97a850be66?pvs=4#13604e4d9e6b80b09a18d4f71a89f1c8
- From compiler's perspective using `constexpr char[]` and `constexpr std::string_view` won't make a difference: check [here](https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,selection:(endColumn:2,endLineNumber:19,positionColumn:2,positionLineNumber:19,selectionStartColumn:2,selectionStartLineNumber:19,startColumn:2,startLineNumber:19),source:'//+Online+C%2B%2B+compiler+to+run+C%2B%2B+program+online%0A%23include+%3Ciostream%3E%0A%23include+%3Cstring%3E%0A%23include+%3Cstring_view%3E%0A%23include+%3Ccstddef%3E%0A%0Asize_t+foo(std::string+const%26+s)+%7B%0A++++return+s.size()%3B%0A%7D%0A%0Aint+main()+%7B++++%0A++++constexpr+std::string_view+sv%7B%22Hi%22%7D%3B%0A++++constexpr+char+cstr%5B%5D%7B%22Hi%22%7D%3B%0A++++%0A++++size_t+const+a%7Bfoo(std::string%7Bsv%7D)%7D%3B%0A++++size_t+const+b%7Bfoo(cstr)%7D%3B%0A%0A++++return+a+-+b%3B%0A%7D'),l:'5',n:'0',o:'C%2B%2B+source+%231',t:'0')),k:50,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:compiler,i:(compiler:g142,filters:(b:'0',binary:'1',binaryObject:'1',commentOnly:'0',debugCalls:'1',demangle:'0',directives:'0',execute:'1',intel:'0',libraryCode:'0',trim:'1',verboseDemangling:'0'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,libs:!(),options:'-O2',overrides:!(),selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1),l:'5',n:'0',o:'+x86-64+gcc+14.2+(Editor+%231)',t:'0')),k:50,l:'4',n:'0',o:'',s:0,t:'0')),l:'2',n:'0',o:'',t:'0')),version:4)